docs/src/components/action/NpmCopyButton.tsx

```diff
@@ -8,6 +8,7 @@ import CheckRounded from '@mui/icons-material/CheckRounded';
 
 const StyledButton = styled(Button)`
```

Comment:
This naming feels off, and see below how this might help to create a foot gun. I think we would be much clearer with, or equivalent:

```jsx
const Button = styled(ButtonUnstyled)`
```

Shadcn has the same problem, can't call their element "styled", they avoid the name conflict like this:

```jsx
import * as AlertDialogPrimitive from "@radix-ui/react-alert-dialog"
```

So in https://mui.com/base-ui/react-button/

```jsx
const CustomButton = styled(Button)(
```

is not going to fly IMHO.

Reply:
How do I do that, though? Would it work to import the Button from Base UI as:

```
import ButtonUnstyled, { ButtonProps } from '@mui/base/Button';
```

Reply:
Correct. I linked this  comment in https://github.com/mui/material-ui/pull/38157#discussion_r1282488480 too, to further test the idea out.